soc/arm/st_stm32/stm32h7/sections.ld

```diff
@@ -16,3 +16,12 @@ SECTION_DATA_PROLOGUE(eth_stm32,(NOLOAD),)
 } GROUP_DATA_LINK_IN(LINKER_DT_NODE_REGION_NAME(DT_NODELABEL(sram3)), LINKER_DT_NODE_REGION_NAME(DT_NODELABEL(sram3)))
 
 #endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(sram4), okay)
+SECTION_PROLOGUE (sram4, ABSOLUTE(DT_REG_ADDR(DT_NODELABEL(sram4))) (NOLOAD),)
```

Comment:
When adding the section for the SRAM4 to map BDMA buffers, are the clock and power already present to supply the D3 domain ?

Reply:
Not sure what you mean? Adding this linker section will not change the order in SRAM sections or this driver is initialized. And the first time the BDMA knows about SRAM4 is during `INIT_LEVEL_APPLICATION`, and by then everything is obviously started up as expected.